php/php.editor/tools/ASTPHP5Parser.cup

```diff
@@ -2419,6 +2547,13 @@ is_variadic:isVariadic expr_without_variable:var
     RESULT = paramsList;
 :}
 
+| error:expr
+{:
+    List paramsList = new LinkedList();
+    paramsList.add(new ASTErrorExpression(exprleft, exprright));
+    RESULT = paramsList;
+:}
+
```

Comment:
To avoid `ASTError`, add `ASTErrorExpression`. 
e.g. 
If we don't add `ASTErrorExpression`, the following (whole statement)is recognized as `ASTError`. In such case, some features(e.g. formatting, CC) may not work correctly. Plus, we have to sanitize the code and parse it again.
```
#[A(1, self::)]
class Test {}
```

Reply:
Great.